source/hwIo/ioThread.py

```diff
@@ -0,0 +1,91 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2016-2022 NV Access Limited, Joseph Lee, Babbage B.V., Davy Kager, Bram Duvigneau,
+# Leonard de Ruijter
+
+import ctypes
+import ctypes.wintypes
+import threading
+import winKernel
+import typing
+from logHandler import log
+import extensionPoints
+import uuid
+
+pre_IoThreadStop = extensionPoints.Action()
+"""
+Executed when the i/o thread is to be stopped.
+This allows components and add-ons to clean up or reset state before background thread shut down.
+Handlers are called with one argument.
+@param ioThread: The thread to shut down
+@type ioThread: IoThread
+"""
+
+
+class IoThread:
+	"""A thread used for background writes and raw I/O, e.g. for braille displays.
+	"""
+
+	thread: typing.Optional[threading.Thread] = None
+	exit: bool = False
+	_apcReferences: typing.Dict[uuid.UUID, winKernel.PAPCFUNC]
+
+	def __init__(self):
+		self._apcReferences = dict()
+
+	def start(self):
+		if self.thread:
+			return
```

Comment:
It might be clearer if this class inherited from `threading.Thread` and the lifecycle of `IoThread` was managed by some owner.

Reply:
Thanks, good idea. I applied that suggestion.